pkg/controller/dnszone/azureactuator.go

```diff
@@ -73,6 +73,10 @@ func (a *AzureActuator) Create() error {
 
 	logger.Debug("Managed zone successfully created")
 	a.managedZone = &managedZone
+	if err := a.ModifyStatus(); err != nil {
```

Comment:
This is the last actuator to do `ModifyStatus` in the `Create` and `Refresh` functions. Can we make `ModifyStatus` private now and remove the call from the dnszone controller?